File: server/router/jxc/stockcheck.go
package jxc

import (
	"github.com/gin-gonic/gin"
)

// StockCheckRouter 盘点中心路由
type StockCheckRouter struct{}

func (r *StockCheckRouter) InitStockCheckRouter(Router *gin.RouterGroup) {
	readOnlyGroup := Router.Group("jxc")
	recordGroup := Router.Group("jxc")
	{
		readOnlyGroup.GET("stockcheck/page", stockCheckApi.GetStockCheckPage)
		readOnlyGroup.GET("stockcheck/detail", stockCheckApi.GetStockCheckDetail)
	}
	{
		recordGroup.POST("stockcheck", stockCheckApi.CreateStockCheck)
		recordGroup.PUT("stockcheck/items", stockCheckApi.UpdateStockCheckItems)
		recordGroup.PUT("stockcheck/complete", stockCheckApi.CompleteStockCheck)
		recordGroup.PUT("stockcheck/cancel", stockCheckApi.CancelStockCheck)
	}
}

File: web/src/api/jxc/stockcheck.js
import service from '@/utils/request'

// 盘点单
export const getStockCheckPage = (params) => service.get('/jxc/stockcheck/page', { params })
export const getStockCheckDetail = (id) => service.get('/jxc/stockcheck/detail', { params: { id } })
export const createStockCheck = (data) => service.post('/jxc/stockcheck', data)
export const updateStockCheckItems = (data) => service.put('/jxc/stockcheck/items', data)
export const completeStockCheck = (id) => service.put('/jxc/stockcheck/complete', { id })
export const cancelStockCheck = (id) => service.put('/jxc/stockcheck/cancel', { id })
